filebeat/registrar/registrar.go

```diff
@@ -269,45 +115,116 @@ func (r *Registrar) Start() error {
 	}
 
 	r.wg.Add(1)
-	go r.Run()
+	go func() {
+		defer r.wg.Done()
+		r.Run()
+	}()
 
 	return nil
 }
 
+// Stop stops the registry. It waits until Run function finished.
+func (r *Registrar) Stop() {
+	logp.Info("Stopping Registrar")
+
+	close(r.done)
+	r.wg.Wait()
+	logp.Info("Registrar stopped")
```

Comment:
Nit: move this function after the first log line and add `defer` just like it is done in `Run`.